cxx/distributions/normal.cc

```diff
@@ -47,24 +44,21 @@ void Normal::posterior_hypers(double* mprime, double* sprime) const {
 
 double Normal::logp(const double& x) const {
   // Based on equation (13) of GaussianInverseGamma.pdf
-  double unused_mprime, sprime;
+  double unused_mprime, sprime2;
   const_cast<Normal*>(this)->incorporate(x);
-  posterior_hypers(&unused_mprime, &sprime);
-  const_cast<Normal*>(this)->unincorporate(x);
-  double sprime2;
   posterior_hypers(&unused_mprime, &sprime2);
-  return -0.5 * log(M_2PI)
-      + logZ(r + N + 1, v + N + 1, sprime)
-      - logZ(r + N, v + N, sprime2);
+  const_cast<Normal*>(this)->unincorporate(x);
+  double sprime;
```

Comment:
Thanks for finding this bug!  Given the variable names and logic, it might make more sense to calculate sprime before the incorporate/unincorporate calls.

Reply:
Done.